app/controllers/api/v1/comments_controller.rb

```diff
@@ -0,0 +1,46 @@
+class Api::V1::CommentsController < ApplicationController
+  def index
+    @comments = Comment.where(post_id: post.id)
+    render json: @comments
+  end
+
+  def show
+    render json: @comment
+  end
+
+  def create
+    @comment = Comment.new(comment_params)
+
+    if @comment.save
+      render json: @comment, status: :created, location: @comment
+    else
+      render json: @comment.errors, status: :unprocessable_entity
+    end
+  end
```

Comment:
- [ Required ] I think it would be nice if you ensure that this method works as expected. Currently, when I try to add a `comment` to a `post` via Insomnia REST client through a `POST` request to the route `http://127.0.0.1:3000/api/v1/users/1/posts/1/comments`, I get the error shown below in the screenshot. I think it would be nice if you ensure that this method works the same way as the `create` method in `app/controllers/comments.rb`.

## Error when creating a comment
![Screenshot from 2023-09-02 14-19-55](https://github.com/Enning94/Blog_App/assets/16392046/ce44d8cb-c58f-46f8-9173-2ce6318a20d1)

Reply:
Kindly please provide further insight on what this error means and how we can work around it. We are stuck implementing the requested changes. @Felix45

Reply:
Hi @Enning94 

I hope you are well, I think that your application currently, does not work correctly because the `create` method in the `api/v1/comments` controller is missing some parameters which are not being passed in. So I would recommend refactoring it as shown in the example below. This is only a suggestion, in case you find another solution feel free to implement it.

## Example code
> api/v1/comments_controller.rb

```ruby
  class Api::V1::CommentsController < ActionController::API # Extend API class which is the recommended base class for API's
  def index
    @comments = Comment.where(post_id: post.id)
    render json: @comments
  end

  def show
    render json: @comment
  end

  def create
    @post    = post # Select the post to be commented on
    @comment = @post.comments.new(comment_params) 

    if @comment.save
      render json: @comment, status: :created
    else
      render json: @comment.errors, status: :unprocessable_entity
    end
  end

  def update
    if @comment.update(comment_params)
      render json: @comment
    else
      render json: @comment.errors, status: :unprocessable_entity
    end
  end

  def destroy
    @comment.destroy
  end

  private

  def comment_params
    params.require(:comment).permit(:text, :author_id) # pass comment text and author in the body of the request
  end

  def user
    @user = User.find(params[:user_id])
  end

  def post
    @post = Post.find(params[:post_id])
  end
end
```

## Results after changes
![Screenshot from 2023-09-02 19-39-47](https://github.com/Enning94/Blog_App/assets/16392046/58e2e1c2-a347-412d-b91e-212fe8ffe84d)

Reply:
@Felix45  Thank you for the insightful feedback, It's very useful.